api/src/main/java/edu/cornell/mannlib/vitro/webapp/modules/rdfDelta/RDFDeltaDatasetFactory.java

```diff
@@ -0,0 +1,92 @@
+
+import edu.cornell.mannlib.vitro.webapp.modules.Application;
+import edu.cornell.mannlib.vitro.webapp.modules.ComponentStartupStatus;
+import edu.cornell.mannlib.vitro.webapp.modules.messaging.JMSMessagingClient;
+import edu.cornell.mannlib.vitro.webapp.utils.configuration.Property;
+
+public class RDFDeltaDatasetFactory {
+
+    private static final Log log = LogFactory.getLog(RDFDeltaDatasetFactory.class);
+
+    private String deltaServerURL;
+
+    private String deltaClientZone;
+
+    private Zone zone;
+
+    private DeltaLink deltaLink;
+
+    private DeltaClient deltaClient;
+
+    private JMSMessagingClient jmsMessagingClient;
+
+    @Property(uri = "http://vitro.mannlib.cornell.edu/ns/vitro/ApplicationSetup#hasDeltaServerURL", minOccurs = 1, maxOccurs = 1)
+    public void setDeltaServerURL(String url) {
+        this.deltaServerURL = url;
+    }
+
+    @Property(uri = "http://vitro.mannlib.cornell.edu/ns/vitro/ApplicationSetup#hasDeltaClientZone", minOccurs = 1, maxOccurs = 1)
+    public void setDeltaClientZone(String zone) {
+        this.deltaClientZone = zone;
+    }
+
+    public void startup(Application application, ComponentStartupStatus ss) {
+        FileOps.ensureDir(deltaClientZone);
+        FileOps.clearAll(deltaClientZone);
+        zone = Zone.connect(deltaClientZone);
+        deltaLink = DeltaLinkHTTP.connect(deltaServerURL);
+        deltaClient = DeltaClient.create(zone, deltaLink);
+        jmsMessagingClient = application.getJMSMessagingClient();
+        log.info(String.format("DeltaClient connected to DeltaServer at %s with zone %s", deltaServerURL, deltaClientZone));
+    }
+
+    public Dataset wrap(String datasourceName, boolean shouldEmitMessage, Dataset dataset) {
+        String datasourceURI = String.format("%s/%s", deltaServerURL, datasourceName);
+        Id dsRef = deltaLink.newDataSource(datasourceName, datasourceURI);
+        deltaClient.attachExternal(dsRef, dataset.asDatasetGraph());
+        deltaClient.connect(dsRef, SyncPolicy.TXN_W);
+        try (DeltaConnection dConn = deltaClient.get(dsRef)) {
+            if (shouldEmitMessage && jmsMessagingClient != null) {
+                dConn.addListener(new DeltaLinkListener() {
+                    @Override
+                    public void append(Id dsRef, Version version, RDFPatch patch) {
+                        try {
+                            jmsMessagingClient.send(RDFPatchOps.str(patch));
+                        } catch (JMSException e) {
+                            log.error(e, e);
+                        }
+                    }
+                });
+            } else {
+                if (jmsMessagingClient == null) {
+                    log.warn("Can not emit messaged without message client");
+                }
+            }
+            log.info(String.format("Wrapped dataset using datasource %s with reference %s", datasourceName, dsRef));
+            return dConn.getDataset();
+        } catch (Exception e) {
```

Comment:
I am not seeing any checked Exceptions being thrown in the `try` block. Maybe remove the `catch` block? ..unless you had something in mind.